cost_reports/app.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+# project imports
+import cost_reports.finops as finops
+
+# stdlib imports
+import logging
+import time
+
+# other imports
+import boto3
+
+
+LOG = logging.getLogger(__name__)
+LOG.setLevel(logging.INFO)
+
+ACCOUNT_TSV_S3_KEY = 'account_cost_centers/table.csv'
+CUR_S3_BUCKET = 'sagebase-cost-reports'
+CUR_S3_PREFIX = 'sagebase'
+REPORT_NAME = 'monthly-costs'
+
+
+class App:
+    s3 = None
+
+    def __init__(self):
+        if App.s3 is None:
+            App.s3 = boto3.client('s3')
+
+        self.account_csv = None
+        self.cur_parquet = None
+        self.report = None
+
+        self.year_month = self._last_month()
+
+    def _last_month(self):
+        """What month was it last month?
+
+        Return a string in the formart `year=XXXX/month=XX` representing
+        the previous month."""
+
+        # format syntax: https://docs.python.org/3.9/library/time.html?highlight=time.strftime#time.strftime
+        year = int(time.strftime("%Y"))
+        month = int(time.strftime("%m")) -  1   # last month
+        return f"year={year}/month={month}"
+
+    def _build_cur_key(self):
+        """Build the path to the parquet file for the current month"""
+
+        s3_path = f"{CUR_S3_PREFIX}/{REPORT_NAME}/{REPORT_NAME}/{self.year_month}"
+
+        # TODO: check for file indicies other than '00001'
+        s3_file = f"{REPORT_NAME}-00001.snappy.parquet"
```

Comment:
this name seems kinda odd. is this name generated by AWS?

Reply:
yeah, this is the name AWS Billing came up with